File: src/main/java/com/github/ccguyka/showupdates/producer/ProjectUpdatesSource.java
package com.github.ccguyka.showupdates.producer;

import com.github.ccguyka.showupdates.objects.ProjectUpdates;

/**
 * Find all the updates of this project.
 */
public class ProjectUpdatesSource {

    private final ParentUpdateSource parentUpdate;
    private final DependencyUpdatesSource dependencyUpdates;
    private final PluginUpdatesSource pluginUpdates;
    private final DependencyManagementUpdatesSource dependencyManagementUpdatesSource;

    public ProjectUpdatesSource(ParentUpdateSource parentUpdate,
            DependencyUpdatesSource dependencyUpdates, PluginUpdatesSource pluginUpdates,
            DependencyManagementUpdatesSource dependencyManagementUpdatesSource) {
        this.parentUpdate = parentUpdate;
        this.dependencyUpdates = dependencyUpdates;
        this.pluginUpdates = pluginUpdates;
        this.dependencyManagementUpdatesSource = dependencyManagementUpdatesSource;
    }

    public ProjectUpdates getProjectUpdates() {
        return ProjectUpdates.builder()
                .withParent(parentUpdate.getUpdate())
                .withDependency(dependencyUpdates.getUpdates())
                .withPlugin(pluginUpdates.getUpdates())
                .withDependencyManagement(dependencyManagementUpdatesSource.getUpdates())
                .build();
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/producer/ProjectUpdatesSource.kt
package com.github.ccguyka.showupdates.producer

import com.github.ccguyka.showupdates.objects.ProjectUpdates
import com.github.ccguyka.showupdates.objects.ProjectUpdates.Companion.builder

/**
 * Find all the updates of this project.
 */
class ProjectUpdatesSource(private val parentUpdate: ParentUpdateSource,
                           private val dependencyUpdates: DependencyUpdatesSource,
                           private val pluginUpdates: PluginUpdatesSource,
                           private val dependencyManagementUpdatesSource: DependencyManagementUpdatesSource) {

    val projectUpdates: ProjectUpdates
        get() = builder()
                .withParent(parentUpdate.update)
                .withDependency(dependencyUpdates.updates)
                .withPlugin(pluginUpdates.updates)
                .withDependencyManagement(dependencyManagementUpdatesSource.updates)
                .build()
}
